README.md

```diff
@@ -74,8 +74,93 @@ require_once __DIR__ . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'a
 $runner->addExtension(new \mageekguy\atoum\blackfire\extension());
 ```
 
+## Other examples
+
+### Define custom metrics
+
+```php
+$this
+    ->blackfire()
+        ->defineMetric(new \Blackfire\Profile\Metric("example_method_calls", "=Example::method"))
+        ->assert("metrics.example_method_calls.count < 10")
+        ->profile(function() {
+            $testedClass = new TestedClass();
+            for ($i = 0; $i < 8; $i++) {
+                $testedClass->method();
+            }
+        })
+;
+```
+
+You can learn more about this on the [custom metric](https://blackfire.io/docs/reference-guide/metrics#custom-metrics)'s section of Blackfire documentation.
+
+### Pass your own configuration
+
+```php
+$this
+    ->given(
+        $profileConfiguration = new \Blackfire\Profile\Configuration(),
+        $profileConfiguration->setTitle('Profile configuration title'),
+        $testedClass = new TestedClass()
+    )
+    ->blackfire()
+        ->setConfiguration($profileConfiguration)
+        ->assert("main.peak_memory < 10mb")
+        ->profile(function() use ($testedClass) {
+            $testedClass->method();
+        })
+;
+```
+
+You can learn more about this on the [profile basic configurable](https://blackfire.io/docs/reference-guide/php-sdk#profile-basic-configuration)'s section of Blackfire documentation.
+
+
 ## Test filtering
 
-If you need to run the tests without the blackfire extension, you can use [atoum's tags](http://docs.atoum.org/en/latest/launch_test.html#tags) and the [ruler extension](https://github.com/atoum/ruler-extension).
+To avoid running the test if the blackfire extension is not loaded, you can use the `@extensions` annotation.
+
+```php
+    /**
+     * @extensions blackfire
+     */
+    public function testExemple()
+    {
+        $this
+            ->blackfire()
```

Comment:
I was thinking: Is it possible to have `->blackfire` instead of `->blackfire()`? Like we have for other assertions. Do we create another object behind the scene?